GoogleMapsComponents/Maps/ImageMapType.cs

```diff
@@ -1,11 +1,87 @@
-﻿using System;
+﻿using Microsoft.JSInterop;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 
 namespace GoogleMapsComponents.Maps
 {
-    public class ImageMapType
+    public class ImageMapType : IDisposable
     {
+        private readonly JsObjectRef _jsObjectRef;
+
+        public readonly Dictionary<string, List<MapEventListener>> EventListeners;
+
+        public Guid Guid => _jsObjectRef.Guid;
+        public string Name { get; private set; }
+        GoogleMap map = null;
+        public async static Task<ImageMapType> CreateAsync(IJSRuntime jsRuntime, string baseUrl, int minZoom, int maxZoom, string name, float opacity)
+        {
+            string initOpts = @"{
+                'getTileUrl': (coord, zoom) => {
+                            return '" + baseUrl + @"' + zoom + '/' + coord.x + '/' + coord.y;
+                        },
+                'tileSize': new google.maps.Size(256, 256),
+                'maxZoom': " + maxZoom.ToString() + @",
+                'minZoom': " + minZoom.ToString() + @",
+                'opacity': " + opacity.ToString() + @",
+                'name': '" + name + @"'
+            }";
+
+            var jsObjectRef = await JsObjectRef.CreateAsync(jsRuntime, "google.maps.ImageMapType", initOpts);
+            var to = new ImageMapType(jsObjectRef) {
+                Name = name
+            };
+            return to;
+        }
```

Comment:
this form of getTileUrl is a typical scenario; however, if we could figure a way to set this up with a ImageMapOptions so that a user could really customize their Url, that would be good, I think.